include/graphene-simd4f.h

```diff
@@ -854,15 +860,27 @@ typedef int graphene_simd4i_t __attribute__((vector_size (16)));
     (graphene_simd4f_t) { __val, __val, __val, __val }; \
   }))
 
+#if defined (HAVE_IEEE_754_FLOAT_DIVISION)
```

Comment:
This cannot work.

You're assuming Graphene's `config.h` is going to be included before `graphene-simd4f.h`, which is not going to be the case for projects that *use* Graphene.

You will need to do something like what we do for `GRAPHENE_HAS_SSE4`: the symbol must be defined in `graphene-config.h`; of course, since you determine the capability at compile time, this might not be doable.

You have three options:

 1. you check for the C standard version in use *and* the compiler in use
 2. you always use the fallback implementation
 3. you add a dependency on a `#define GRAPHENE_NO_FLOAT_DIV`, like we have a `GRAPHENE_SIMD_BENCHMARK` symbol that forces the SIMD implementation; projects that require some niche compiler can inject that symbol before including `graphene.h` and get the fallback implementation

Reply:
Of course, you're right, it's a header. Unfortunately there's no specific macro or definition that can be relied upon in compile-time to detect this. I'll see what I can do.

Reply:
@ebassi, since there was no apparent performance benefit from allowing division by zero in the GCC vectorization implementation, I went with (2) and dropped it.

I left the optimization only for the scalar implementation which is part of `libgraphene`'s compilation unit.